src/shared/services/find-replace-all.service.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@angular/core';
+import { List, Map } from 'immutable';
+
+@Injectable()
+export class FindReplaceAllService {
+
+  // TODO: fix compiler errors when type of immutable `List<any> | Map<string, any>`
+  // TODO: add doc string
+  findReplaceInImmutable(immutable: any, find: string, replace: string, matchWhole = false, diffHtml?: Array<any> | Object):
+    { replaced: any, diffHtml: Array<any> | Object } {
+    let immutableAsMutable = immutable.asMutable();
+    // create empty array or object for the immutable to store store diff
+    diffHtml = List.isList(immutable) ? [] : {};
+    immutableAsMutable.forEach((value, key) => {
+      if (typeof value === 'string') {
+        // assign value to diff as initial, if nothing has changed it will remain same
+        let diff = value;
+        // create html diff for each possible change
+        let singleDiffHtml = `<strong style='color: green;'>${replace}</strong><del><em style='color: red;'>${find}</em></del>`;
```

Comment:
I was thinking the same but then I had to add this class to somewhere global, so that's why I wasn't sure to add it or hardcode it in string.